### PR TITLE
Add curl, ruby and python examples

### PR DESCRIPTION
**Pull request by jvlcek**
Our documentation, much like most REST API documentation, illustrates the supported REST API with the URI and example payloads in a language neutral fashion. Although complete it can be difficult for a new user to learn how to actually apply these language neutral illustrations to real world usage.

This PR attempts to bridge this learning gap by documenting the same three examples, a GET, a POST with a simple payload and a GET with filtering, and to illustrate each using cURL, Ruby and Python. This will likely go a long way in helping new users ramp up quickly with the ManageIQ REST API.

### CODE REVIEW

**Comment by jvlcek** [comment]
@abellotti Please review.

**Comment by jvlcek** [comment]
@abellotti Changes made as we discussed. Should be good-to-go now. Thank you for your help.

File: api/overview/usage_examples.adoc [hunk 1]
```diff
@@ -0,0 +1,479 @@
...
+    "settings": null,
+    "tenant_id": "2",
+    "updated_on": "2019-06-19T19:15:51Z"
+}
+----
+
+[[curl-example3]]
+==== Example 3: A **GET** using filtering with cURL
+
+Filtering can be done by passing arguments in the URL. Arguments are preceded by the **&** character, which can cause problems as the shell interprets the **&** character as an instruction to: **run this command in the background**. To avoid this the arguments can instead be passed to curl using the **--get** flag combined with each argument being passed with the **--data** flags.
```
Comment (abellotti) [inline api/overview/usage_examples.adoc]: replace arguments/Arguments with parameters/Parameters  (this is more accurate as these are called query parameters).

File: api/overview/usage_examples.adoc [hunk 2]
```diff
@@ -0,0 +1,479 @@
...
+  - <<python-example3,Example 3: A **GET** using filtering with Python>>
+
+'''''
+
+[[curl]]
+== cURL
+
+The cURL command can be used from the shell. However the shell's interpretation of some characters, for example **&**, can make it awkward to use for complex cases.
+To help alleviate this the payload can be passed to cURL using the **--data** flag, where the payload is stored in a file in the format specified 
+in the content-type, **JSON** in our example.
```
Comment (abellotti) [inline api/overview/usage_examples.adoc]: feels awkward, maybe just "...  the --data flag, where the JSON payload is provided in a file."

Comment (jvlcek) [reply api/overview/usage_examples.adoc]: Agreed. Will do.

File: api/overview/usage_examples.adoc (hunk 1 above)
Comment (jvlcek) [reply api/overview/usage_examples.adoc]: 👍 Will do.

**Comment by jvlcek** [comment]
@abellotti Thank you again. I've made the last couple changes you've requested. Let me know if you see anything else.

**Comment by abellotti** [comment]
Thanks @jvlcek for providing and updating this new API doc page.

**Pull request merged**